src/items/controllers/item.controller.ts

```diff
@@ -4,18 +4,32 @@ import { Public } from '../../auth/decorators/public.decorator';
 import { BuyRequestDto } from '../dto/buy-request.dto';
 import { ItemRequestDto } from '../dto/item-request.dto';
 import { ListItemRequestDto } from '../dto/list-item-request.dto';
+import { History } from '../entities/history.entity';
 import { Item } from '../entities/item.entity';
+import { HistoryService } from '../services/history.service';
 import { ItemService } from '../services/item.service';
 @Controller('items')
 export class ItemsController {
-  constructor(private itemService: ItemService) {}
+  constructor(private itemService: ItemService, private historyService: HistoryService) {}
 
   @Public()
   @Get()
   findAll(): Promise<Item[]> {
     return this.itemService.findAll();
   }
 
+  @Public()
+  @Get('/history')
+  findAllHistory(): Promise<History[]> {
+    return this.historyService.findAll();
```

Comment:
I wonder if we need this endpoint? My concern with this is that we are retrieving all the history (no restrictions, not filtering, no pagination) from the db. This is dangerous for a production env